_security/authentication-backends/saml.md

```diff
@@ -224,7 +224,7 @@ SAML, unlike other protocols, is not meant to be used for exchanging user creden
 
 Name | Description
 :--- | :---
-`exchange_key` | The key to sign the token. The algorithm is HMAC256, so it should have at least 32 characters.
+`exchange_key` | The key to sign the token. The algorithm is HMAC-SHA512, which mixes the `exchange_key` with the message data and produces an output hash that is 512 bits/64 bytes in length. The `exchange_key` must have a value and a `base64` URL encoding. 
```

Comment:
What type of value should the `exchange_key` have? The output hash?

```suggestion
`exchange_key` | The key to sign the token. The algorithm is HMAC-SHA512, which mixes the `exchange_key` with the message data and produces an output hash that is 512 bits/64 bytes in length. The `exchange_key` must have a value and a `base64` URL encoding. 
```

Reply:
Let me check on that Nate, thanks.

Reply:
@natebower A quick update: the value that you enter for the exchange key would be characters like this: 1a2a3a4a5a6a7a8a9a0a1b2b3b4b5b6b
But when it is used, the output hash that's generated is made up of 64bytes.
Let me just check on a couple of other things though!

Reply:
> @natebower A quick update: the value that you enter for the exchange key would be characters like this: 1a2a3a4a5a6a7a8a9a0a1b2b3b4b5b6b But when it is used, the output hash that's generated is made up of 64bytes. Let me just check on a couple of other things though!

@Naarcha-AWS

Reply:
So would "hash value" make sense here @john-eliatra?

Reply:
Yes, hash value would make sense @Naarcha-AWS. I just have another couple of things to check actually, whether it's just HMAC-SHA512 algorithm or is HMAC256 still applicable. Overall it determines what to say to the user about maximum number of characters that can be entered for the exchange key. Thx, John